db/migrate/20150510022120_add_client_fields_to_proposal.rb

```diff
@@ -0,0 +1,6 @@
+class AddClientFieldsToProposal < ActiveRecord::Migration
+  def change
+    add_column :proposals, :client_fields, :json
+    add_column :proposals, :subclass, :string
```

Comment:
I'd use a `type` column to have Rails handle the subclass conversion automatically.

http://api.rubyonrails.org/classes/ActiveRecord/Inheritance.html